godot-core/src/engine/gfile.rs

```diff
@@ -0,0 +1,823 @@
+                        &path,
+                        flags,
+                        FileAccess::get_open_error()
+                    ),
+                )
+            })?;
+
+        Ok(Self::from_inner(fa))
+    }
+
+    /// Open a file encrypted by byte key
+    ///
+    /// Opens a file encrypted by 32-byte long [`PackedByteArray`] located at
+    /// `path`, creating new [`GFile`] object.
+    /// For [`ModeFlags`] description check the [`GFile`] documentation.
+    pub fn open_encrypted(
+        path: impl Into<GString>,
+        flags: ModeFlags,
+        key: PackedByteArray,
+    ) -> IoResult<Self> {
+        let path: GString = path.into();
+        let fa = FileAccess::open_encrypted(path.clone(), flags, key).ok_or_else(|| {
+            IoError::new(
+                ErrorKind::Other,
+                format!(
+                    "can't open file {} in mode {:?}; GodotError: {:?}",
+                    &path,
+                    flags,
+                    FileAccess::get_open_error()
+                ),
+            )
+        })?;
+
+        Ok(Self::from_inner(fa))
+    }
+
+    /// Open a file encrypted by password
+    ///
+    /// Opens a file encrypted by a `pass` located at `path`, creating new
+    /// [`GFile`] object. For [`ModeFlags`] description check the [`GFile`]
+    /// documentation.
+    pub fn open_encrypted_with_pass(
+        path: impl Into<GString>,
+        flags: ModeFlags,
+        pass: GString,
+    ) -> IoResult<Self> {
+        let path: GString = path.into();
+        let fa =
+            FileAccess::open_encrypted_with_pass(path.clone(), flags, pass).ok_or_else(|| {
+                IoError::new(
+                    ErrorKind::Other,
+                    format!(
+                        "can't open file {} in mode {:?}; GodotError: {:?}",
+                        &path,
+                        flags,
+                        FileAccess::get_open_error()
+                    ),
+                )
+            })?;
+        Ok(Self::from_inner(fa))
+    }
+
+    /// Creates new [`GFile`] from provided [`Gd`]<[`FileAccess`]>.
+    ///
+    /// For this method to work, the provided `file_access` must be unique - no
+    /// other symbol can reference it. Its state is retained: [`ModeFlags`] with
+    /// which it was open and current internal cursor position.
+    pub fn from_unique(file_access: Gd<FileAccess>) -> Result<Self, NotUniqueError> {
+        let file_access = NotUniqueError::check(file_access)?;
+        Ok(Self::from_inner(file_access))
+    }
+
+    /// Retrieve inner [`Gd`]<[`FileAccess`]>
+    ///
+    /// Deconstructs the [`GFile`] intance and retrieves the inner
+    /// [`Gd`]<[`FileAccess`]>. Useful when the [`Gd`]<[`FileAccess`]> is needed
+    /// in some other function or method, and you would like to preserve its
+    /// state or just keep the file open while transferring.
+    pub fn into_inner(self) -> Gd<FileAccess> {
+        self.fa
+    }
```

Comment:
Brief doc could be more explicit:
```rs
/// Converts this `GFile` into a raw Godot `FileAccess`.
/// 
/// This instance will be destroyed, but the file is kept open as long as there is at least one
/// reference pointing to it.
```
The "useful when the return type is needed in some other function or method" is somewhat obvious, you can omit it :slightly_smiling_face: